### DISCUSSION TITLE
Error messages on startup ALT-Scann8

### DISCUSSION THREAD

**Discussion by ToRa-51** [jareff-g/ALT-Scann8#36] [Q&A] [answered]
Hello Juan, 

Below is a table with the versions of the software (ALT-Scann8-UserInterface and ALT-Scann8-Controller with date on Github) which I downloaded and tried to test.
![Tabel software ALT-Scann8](https://github.com/jareff-g/ALT-Scann8/assets/26844127/f904981c-4e09-44ee-84df-89e599caa72e)

The first three versions through 14 Oct. ’23 were good and the scanner was already doing well, although I didn't know how to set everything up at the time. (Images on the video [date 18 Nov. '23] are a bit dark, possibly because it was filmed with backlight.)
[https://youtu.be/zagrWB_oMYE]

Made a log file of this movie the day before yesterday. Don't pay attention to the number with scanned frames, I started the same movie again.
[ALT-Scann8.20231125.log](https://github.com/jareff-g/ALT-Scann8/files/13467371/ALT-Scann8.20231125.log)

After starting ALT-Scann8, a window 'Persisted session data exist' appears, here 'Yes' or 'No' must be clicked. With the 26 Oct. '23 version, clicking 'Yes' will appear the following on the terminal window. After this, the program is fixed.
![IMG_7555](https://github.com/jareff-g/ALT-Scann8/assets/26844127/09b4bd20-472e-4ac3-8c8c-377f5e31e80f)

And the same, but on 'No'... Even then, the program is fixed and can only be closed by closing the terminal window.
![IMG_7556](https://github.com/jareff-g/ALT-Scann8/assets/26844127/823364bf-1a95-4469-9ba4-1c014fa1121d)

All subsequent versions through 16 Nov. '23 do the same in both situations. Click on 'Yes'...
![IMG_7548](https://github.com/jareff-g/ALT-Scann8/assets/26844127/3d450951-34cc-43fb-b9c5-13c2fe82d208)

And click 'No'....
![IMG_7549](https://github.com/jareff-g/ALT-Scann8/assets/26844127/ddb12e46-80a0-442a-959a-b5fae9edd590)

I hope that you have enough information to solve it for me, so that I can also use all the following versions on my scanner until 23 Nov. '23.

Greetings, 
Ton.

PS 1: I will still try to create a log file with the versions that give error messages after startup. I'll upload it here later.

PS 2: My thanks are great for developing this alternative software for the beautifully designed 'T-Scann 8' scanner by Torulf Holmström.

**Comment by jareff-g** [comment, marked as answer]
Hello Ton,

Thanks for your detailed report. I think it has allowed me to find the root causes (there were two problems), that should be fixed in the version i just committed. So, no need to create additional log files, just download the latest version, test it, and let me know

The problems were due to the UI homogenization I did mid-October, where I removed/updated several widgets, but did not remove all references to the ones that disappeared. Since Python only checks that at run time, and my configuration was not causing the affected code to run, I did not experience those problems, but you did, due to your specific configuration.

Just one comment: You have automatic white balance activated (this triggered one of the errors) which normally is considered (from what I have read, I am no expert) to be wrong, as you might have weird color variations. Since then I always use what Torulf recommends for his version, 2.2 for both red and blue WB. But this is a matter of personal taste, do as you like best.

Regards,
Juan

PS: Thanks for using ALT-Scann8, having other people using it allows to discover more bugs (well, I could do a wider test campaign, with automatic tests and all, but don't have the time).

PS2: Next time you can open an issue. It will ask you for a lot of information, but feel free to skip any parts you do not have info about.
**
![image](https://github.com/jareff-g/ALT-Scann8/assets/98735354/c265e018-2e18-4b30-870d-ea23b1610ca9)
**

**Reply by ToRa-51** [reply]
Juan, 

Unfortunately lately I didn't have much time to continue with my hobby project, but luckily I was able to work on that for a few hours yesterday. 
The last version I installed is 1.8.4 and at least it turned out that the two error messages were no longer there during startup. 

What I have seen is that the saved frames are not all equally sharp. Pay attention to the white stripes on the chair. Is there anything else to be done about this?
https://github.com/jareff-g/ALT-Scann8/assets/26844127/b74a9002-9647-4270-ab2e-06f7d98121b7

Furthermore, when closing ALT-Scann8 in the terminal window I saw some error notifications? Is this true?
![IMG_7623](https://github.com/jareff-g/ALT-Scann8/assets/26844127/f7c42efd-6adf-4e22-9566-0da80c1bf5fd)

Attached is also the log file of the scan session with version 1.8.4. 
I noticed that line 56 (file opened in Geany on Raspberry Pi) says: No OpenGL_accelerate module loaded: No module named 'OpenGL_accelerate' and then there are a lot of lines in which OpenGL occurs a lot of times. Is this okay or do I have to install something on the Raspberry Pi 4B myself?￼
[ALT-Scann8.20231209.log](https://github.com/jareff-g/ALT-Scann8/files/13628304/ALT-Scann8.20231209.log)

So much for my questions after yesterday's test. Undoubtedly there will be some more questions.

Greetings, Ton.

**Reply by jareff-g** [reply]
Hello Ton,

No problem with your dedication, a hobby is just a hobby :-)

- Frames not equally sharp: From what I see on the video this seems to be caused by film not being completelly steady at the time the frame was captured. To prevent that you can use the stabilization delay parameter in the user interface, that you can increase to make sure film does not move when the snapshot is taken. Personally I use 80 ms here, but might depend on each specific scanner. **Try to increase this value**.
![image](https://github.com/jareff-g/ALT-Scann8/assets/98735354/9996e929-808a-4d04-bcd8-2ae29eeacd95)
- Errors on exit: I already noticed those, but since they happen only in exit, and the error is thrown by the PiCamera2 module. I did no pay much attention to it. Will try to spend some time on it next time I use my scanner.
- OpenGL error: I have no idea on this one, never saw it before. Do you notice any side effect while scanning? Same thing, next time I start my scanner I'll check in my back history to check if I still have the dependencies I installed (I thought I saved them somewhere but cannot find them now).
Rgds/Juan

**Reply by ToRa-51** [reply]
Hai Juan, 

I looked at the entire video which the part with the blurred images is part of and saw that the stabilization delay was still at the same value as when starting the User Interface (250ms). So that couldn't be the cause. Tried to lower this setting again yesterday. The speed of scanning may increase, but the arousing of the film is then very jerky and also makes a lot of noise.
Tomorrow I will look at this again in more specific and then mention it again here.

Ton.

**Reply by jareff-g** [reply]
Hi Ton,

I just looked at the video again and have noticed something interesting. The blurriness noticed in some frames is horizontal, which means means perpendicular to the direction film is moving, so the stability delay, or the scan speed are unlikely to cause that effect. Also, to confirm this assumption, if you look at the frames where the image is blurry, you can see that the image of the sprocket holes on the left, and also the scratches on the film, are not blurry.

So, as far as I can tell with the evidence I see, that blurriness should already exist in the original film. If you have a good magnifying glass, or a microscope, it would be nice if you could confirm that by looking directly at the film, but I'm certain that is the origin. Alternatively, you can perform several scans of that fragment of the film and, if I am correct, the blurriness should always appear in the same frames (contrary to it appearing on random frames if it was resulting from the scan process).

Regarding the noise yes, the scanner can get quite noisy, specially if you increase the speed. I think this is mainly due to the take-up reel, and in the past I have tried (and mostly failed) to reduce it by modifying the take-up algorithm. Where I had more success was in reducing noise while rewind is ongoing, by adding bearings to the rollers that guide the film in that action (link to my mod in thingiverse [here](https://www.thingiverse.com/thing:6329036)).

Rgds/Juan

PS: Even if, as stated above, I think it is unrelated to your issue, recommended values for stable operation are speed=5, stability delay=100 ms

**Reply by ToRa-51** [reply]
Juan,

In recent days I have tried to do some tests to confirm your claim of blurred images.

I have a stereo microscope here, which I use when soldering SMD components or repairs to PCBs with small parts (or removing a splinter in a finger 😃). However, this microscope only has a magnification of 20x, so that the frames cannot actually be properly assessed for not being sharp. 
In another attempt to view the blurred frames, I unplugged the film winding stepper motor. The scanner could now just start up, so this winding motor was not energized. Now it was possible to manually go to the scene in question and via the 'Real Time Display' button I could then view two consecutive frames. There it could be seen that first 4 or 5 frames are sharp and then 2 or 3 were more or less sharp. So this confirms your conclusion that it had to be on the original. I watched another movie this way after this and it showed the same thing, occasionally blurry frames in the scene. In itself, it is not so noticeable, provided there are no vertical lines on the image.

Cause might be with my father's camera?
![IMG_7629](https://github.com/jareff-g/ALT-Scann8/assets/26844127/b982c473-3bf1-4154-a843-16e9b2ed5b91)

Regards and Merry Christmas,
Ton.

**Reply by jareff-g** [reply]
Hello Ton,

Yes, using the real time view was the good troubleshooting choice, I don't know how I missed that. Also, instead of unplugging the winding stepper motor, you could have used the 'movie forward' button to advance the movie while the real time view was activated. That's what I do in similar cases, manually moving the capstan back and forth (of course after activating 'unlock reels') until I find the frame I want.

I have no idea whether the camera could be the origin, in any case still weird that it blurs horizontally.

The camera is a really nice model, looks like coming out of the box for the first time.

Best regards, and merry Christmas to you also,
Juan

**Comment by jareff-g** [comment]
Also, if you have any question about any aspect of ALT-Scann8 that is not detailed in the [wiki](https://github.com/jareff-g/ALT-Scann8/wiki/ALT-Scann-8:-Description-of-UI) let me know, and I'll update it as required.
